test/kube_checks.go

```diff
@@ -121,6 +136,31 @@ func countEndpointsNum(e *corev1.Endpoints) int {
 	return num
 }
 
+// GetEndpointAddresses returns addresses of endpoints for the given service.
+func GetEndpointAddresses(client *KubeClient, svcName, svcNamespace string) ([]string, error) {
+	endpoints, err := client.Kube.CoreV1().Endpoints(svcNamespace).Get(svcName, metav1.GetOptions{})
+	if err != nil || countEndpointsNum(endpoints) == 0 {
+		return nil, fmt.Errorf("no endpoints or error: %w", err)
+	}
+	var hosts []string
+	for _, sub := range endpoints.Subsets {
+		for _, addr := range sub.Addresses {
+			hosts = append(hosts, addr.IP)
+		}
+	}
+	return hosts, nil
+}
+
+// WaitForChangedEndpoints waits until the endpoints for the given service differ from origEndpoints.
+func WaitForChangedEndpoints(client *KubeClient, svcName, svcNamespace string, origEndpoints []string) error {
+	span := logging.GetEmitableSpan(context.Background(), "WaitForChangedEndpoints/"+svcName)
+	defer span.End()
```

Comment:
I wonder why do we need span here?

Reply:
I added that just to be similar to other functions that wait for something, in the same file. Can this be removed?

Reply:
OK. I can remove that. I just thought that since this function is in knative.dev/pkg it could be later used on other places possibly, so added this to align it with other functions.